### PR TITLE
Assignment 2 - Oscar So

### PR DESCRIPTION
**Pull request by oscarso2000**

### CODE REVIEW

File: assignment2.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+1.Yes, definitely. This is because an equation will most likely not only have one operator. For example: (1+5)*6. Furthermore, there are instances where many can be used, for example when finding whether a statement is true of false. The example is given on question 3, using greater than or equal to, less than, not equal to, ANDs, for comparison. the OR comparison can be used as well if necessary ||.
```
Comment (oldclesleycode) [inline assignment2.txt:1]: Correct! 💯

File: assignment2.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+1.Yes, definitely. This is because an equation will most likely not only have one operator. For example: (1+5)*6. Furthermore, there are instances where many can be used, for example when finding whether a statement is true of false. The example is given on question 3, using greater than or equal to, less than, not equal to, ANDs, for comparison. the OR comparison can be used as well if necessary ||. 
+2.Logical errors are the problems created by the outcome of the code, and you will not know where the error is unless you understand the production and the cause of the problem. While syntax errors are similar to careless mistakes in spelling, where the compiler will tell the coder where the small mistake is.
```
Comment (oldclesleycode) [inline assignment2.txt:2]: Correct! 💯

File: assignment2.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+3.True
```
Comment (oldclesleycode) [inline assignment2.txt:3]: Correct, but remember that boolean values technically don't exist in C, so the answer is actually 1!

File: assignment2.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+4.I am guessing this is invalid because this logic has to be split into two separate equations for the compiler to understand. Eg 0<x && x<15 .
```
Comment (oldclesleycode) [inline assignment2.txt:4]: Correct! 💯

File: assignment2.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+5i. There is no "then" in code.
```
Comment (oldclesleycode) [inline assignment2.txt:5]: Correct! 💯

File: assignment2.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+5ii. Parentheses is required around a>b
```
Comment (oldclesleycode) [inline assignment2.txt:6]: Correct! 💯

File: assignment2.txt [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+this is correct, but curly braces are highly recommended to prevent misunderstanding and confusion.
```
Comment (oldclesleycode) [inline assignment2.txt:7]: Perfect! 💯

File: assignment2.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+Adding curly braces is one issue, if not, there should be a semi colon after c=0.i
```
Comment (oldclesleycode) [inline assignment2.txt:8]: Curly braces aren't needed since it's just one line, but you are right about the semicolon!

File: calculator.c [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+		case '+':
+			z = x+y;
+			break;
+		case '-':
+			z = x-y;
+			break;
+		case '*':
+			z = x*y;
+			break;
+		case '/':
```
Comment (oldclesleycode) [inline calculator.c:22]: You didn't include the mod function!

File: calculator.c [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+			z=x/y;
+			break;
+			//		case '%':
+			//			z=x%y;
+			//			break;
+			// does not work because x and y are floats
+		default:
+			printf("Invalid Responses.");
+	}
```
Comment (oldclesleycode) [inline calculator.c:31]: where is your return 0;?

File: randnum.c [hunk 11]
```diff
@@ -0,0 +1,18 @@
+#include<stdio.h>
```
Comment (lesley2958) [inline randnum.c]: Great job! the only suggestion i have is to include text that will prompt the user.

File: positivenegative.c [hunk 12]
```diff
@@ -0,0 +1,14 @@
+//Write a program to check if a user-input number is positive or negative. Assume the user enters a valid number, but that number may not be an integer. File name is your choice, as long as it makes sense.
```
Comment (oldclesleycode) [inline positivenegative.c:1]: Great job! the only suggestion i have is to include text that will prompt the user.

File: randnum.c [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+#include<stdlib.h>
+#include<time.h>
+
+int main(){
+	srand(time(NULL));
+	int r = rand()%11;
```
Comment (oldclesleycode) [inline randnum.c:7]: You were incredibly close! Your range included 0, which it wasn't supposed to do. To fix this, you should have written int r = rand()%10 +1; . Besides that, nice job being on the right track towards an efficient program!

**Pull request closed without merging**